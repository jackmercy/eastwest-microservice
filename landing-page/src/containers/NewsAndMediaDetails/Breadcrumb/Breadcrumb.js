import Link from 'next/link';
import { Fragment, memo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Content } from '../../components';

import useStyles from './breadcrumb.styles';

const Breadcrumb = ({ paths }) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Content className={classNames('hide-scrollbar', classes.content)}>
        {paths.map((path, index) => (
          // eslint-disable-next-line react/no-array-index-key
          <Fragment key={index}>
            {path.href ? (
              <Link href={path.href} passHref>
                <a href={path.href} className={classes.link} title={path.label}>
                  {path.label}
                </a>
              </Link>
            ) : (
              <span
                href={path.href}
                className={classes.subLink}
                title={path.label}
              >
                {path.label}
              </span>
            )}
            {index < paths.length - 1 ? (
              <span className={classes.arrow}>
                <i className="icon-caret-right" />
              </span>
            ) : null}
          </Fragment>
        ))}
      </Content>
    </div>
  );
};
Breadcrumb.propTypes = {
  paths: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      href: PropTypes.string,
    }),
  ).isRequired,
};

export default memo(Breadcrumb);
