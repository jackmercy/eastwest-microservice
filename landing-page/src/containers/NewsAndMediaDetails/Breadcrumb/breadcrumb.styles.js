import { createUseStyles } from 'react-jss';

import COLORS from '../../../constants/COLORS';
import { hexToRgbA, rem } from '../../../utils/jss';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    width: '100%',
    marginBottom: rem(40),
    backgroundColor: hexToRgbA(COLORS.bgBlack01, 0.05),
  },
  content: {
    paddingTop: rem(4),
    paddingBottom: rem(4),
    fontWeight: 600,
    maxWidth: rem(1270),
    fontSize: rem(12),
    lineHeight: rem(16),
    overflow: 'auto',
    display: 'flex',
    alignItems: 'center',
  },
  link: {
    textDecoration: 'none',
    color: COLORS.textBlack01,
    letterSpacing: '0.05em',
    whiteSpace: 'nowrap',
    '&:hover': { color: COLORS.primary },
  },
  arrow: {
    display: 'inline-block',
    margin: [0, rem(10), 0, rem(8)],
    '& > [class^=icon-]': { fontSize: rem(8) },
  },
  subLink: { fontWeight: 400, color: COLORS.textGray, whiteSpace: 'nowrap' },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxMd}px)`]: {
    root: { display: 'none' },
  },
});
