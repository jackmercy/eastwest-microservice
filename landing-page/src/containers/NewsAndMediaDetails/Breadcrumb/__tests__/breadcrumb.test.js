import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Breadcrumb from '../Breadcrumb';

describe('Testing NewsAndMediaDetails > Breadcrumb', () => {
  const props = {
    paths: [
      {
        label: 'News and media',
        href: '/news-and-media',
      },
      {
        label: 'Technology news',
        href: '/news-and-media',
      },
      {
        label:
          // eslint-disable-next-line max-len
          'iPhone SE has some of the best iPhone 11 features, including wireless charging',
      },
    ],
  };
  it('should render match snapshot', () => {
    const wrapper = shallow(<Breadcrumb {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
