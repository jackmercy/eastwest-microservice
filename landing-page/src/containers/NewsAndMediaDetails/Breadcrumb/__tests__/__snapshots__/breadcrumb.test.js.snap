// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NewsAndMediaDetails > Breadcrumb should render match snapshot 1`] = `
<div
  className="root-0-2-1"
>
  <Memo(Content)
    className="hide-scrollbar content-0-2-2"
  >
    <Link
      href="/news-and-media"
      passHref={true}
    >
      <a
        className="link-0-2-3"
        href="/news-and-media"
        title="News and media"
      >
        News and media
      </a>
    </Link>
    <span
      className="arrow-0-2-4"
    >
      <i
        className="icon-caret-right"
      />
    </span>
    <Link
      href="/news-and-media"
      passHref={true}
    >
      <a
        className="link-0-2-3"
        href="/news-and-media"
        title="Technology news"
      >
        Technology news
      </a>
    </Link>
    <span
      className="arrow-0-2-4"
    >
      <i
        className="icon-caret-right"
      />
    </span>
    <span
      className="subLink-0-2-5"
      title="iPhone SE has some of the best iPhone 11 features, including wireless charging"
    >
      iPhone SE has some of the best iPhone 11 features, including wireless charging
    </span>
  </Memo(Content)>
</div>
`;
